src/main/kotlin/org/kamiblue/client/gui/hudgui/elements/world/Radar.kt

```diff
@@ -0,0 +1,146 @@
+) {
+
+    private val radarScale by setting("Radar scale", 3f, 1f..10f, 0.1f)
+
+    /* Entity type settings */
+    private val players = setting("Players", true)
+    private val passive = setting("Passive Mobs", false)
+    private val neutral = setting("Neutral Mobs", true)
+    private val hostile = setting("Hostile Mobs", true)
+    private val invisible = setting("Invisible Entities", true)
+
+    private val backgroundColor by setting("Background color", ColorHolder(28, 28, 28, 200), true)
+    private val chunkGridColor by setting("Chunk grid color", ColorHolder(255, 0, 0, 100), true)
+    private val distantChunkColor by setting("Distant chunk color", ColorHolder(100, 100, 100, 100), true, description = "Chunks that are not in render distance")
+    private val newChunkColor by setting("New chunk color", ColorHolder(255, 0, 0, 100), true)
+
+    override val hudWidth: Float = 130.0f
+    override val hudHeight: Float = 130.0f
+
+    private val radius get() = min(hudWidth, hudHeight) / 2
+
+    override fun renderHud(vertexHelper: VertexHelper) {
+        super.renderHud(vertexHelper)
+        runSafe {
+            drawBorder(vertexHelper)
+            if (NewChunks.isEnabled && NewChunks.renderMode.value != NewChunks.RenderMode.WORLD) drawNewChunks(vertexHelper)
+            drawEntities(vertexHelper)
+            drawLabels()
+        }
+    }
+
+    private fun drawBorder(vertexHelper: VertexHelper) {
+        glTranslated(radius.toDouble(), radius.toDouble(), 0.0)
+        drawCircleFilled(vertexHelper, radius = radius.toDouble(), color = backgroundColor)
+        drawCircleOutline(vertexHelper, radius = radius.toDouble(), lineWidth = 1.8f, color = primaryColor)
+        glRotatef(mc.player.rotationYaw + 180, 0f, 0f, -1f)
+    }
+
+    private fun drawNewChunks(vertexHelper: VertexHelper) {
+        val playerOffset = Vec2d((mc.player.posX - (mc.player.chunkCoordX shl 4)), (mc.player.posZ - (mc.player.chunkCoordZ shl 4)))
+        val chunkDist = (radius * radarScale).toInt() shr 4
+        for (chunkX in -chunkDist..chunkDist) {
+            for (chunkZ in -chunkDist..chunkDist) {
+                val pos0 = getChunkPos(chunkX, chunkZ, playerOffset)
+                val pos1 = getChunkPos(chunkX + 1, chunkZ + 1, playerOffset)
+
+                if (squareInRadius(pos0, pos1)) {
+                    val chunk = mc.world.getChunk(mc.player.chunkCoordX + chunkX, mc.player.chunkCoordZ + chunkZ)
+                    if (!chunk.isLoaded)
+                        drawRectFilled(vertexHelper, pos0, pos1, distantChunkColor)
+                    drawRectOutline(vertexHelper, pos0, pos1, 0.3f, chunkGridColor)
+                }
+            }
+        }
+
+        for (chunk in NewChunks.chunks) {
+            val pos0 = getChunkPos(chunk.x - mc.player.chunkCoordX, chunk.z - mc.player.chunkCoordZ, playerOffset)
+            val pos1 = getChunkPos(chunk.x - mc.player.chunkCoordX + 1, chunk.z - mc.player.chunkCoordZ + 1, playerOffset)
```

Comment:
Extend this on SafeClickEvent instead of calling mc.player, mc.world for null safety in concurrency